ra2ce/analyses/indirect/analyses_indirect.py

```diff
@@ -333,30 +333,24 @@ def multi_link_redundancy(self, graph, analysis):
                     alt_nodes = nx.dijkstra_path(graph, u, v)
 
                     # append to calculation dataframe
-                    df_calculated = df_calculated.append(
-                        {
-                            "u": u,
-                            "v": v,
-                            "rfid": str(edata["rfid"]),
-                            "alt_dist": alt_dist,
-                            "alt_nodes": alt_nodes,
-                            "connected": 1,
-                        },
-                        ignore_index=True,
-                    )
+                    df_calculated = pd.concat([df_calculated, pd.DataFrame({
+                        "u": u,
+                        "v": v,
+                        "rfid": str(edata["rfid"]),
+                        "alt_dist": alt_dist,
+                        "alt_nodes": [alt_nodes],
+                        "connected": 1,
+                    })], ignore_index=True)
                 else:
                     # append to calculation dataframe
-                    df_calculated = df_calculated.append(
-                        {
-                            "u": u,
-                            "v": v,
-                            "rfid": str(edata["rfid"]),
-                            "alt_dist": np.NaN,
-                            "alt_nodes": np.NaN,
-                            "connected": 0,
-                        },
-                        ignore_index=True,
-                    )
+                    df_calculated = pd.concat([df_calculated, pd.DataFrame({
+                        "u": [u],
+                        "v": [v],
+                        "rfid": [str(edata["rfid"])],
+                        "alt_dist": [np.NaN],
+                        "alt_nodes": [np.NaN],
+                        "connected": [0],
```

Comment:
Why are they all here a list but not in the one above? could you please write a comment explaining this?

Reply:
Lists from values, otherwise cannot find any index for the newly created DataFrame. This is added to the code as well